File: src/tf_turtle/src/rand_turtle1.cpp
#include "geometry_msgs/Twist.h"
#include "ros/ros.h"

#include <random>

double generate_random() {
    std::random_device rand;
    std::mt19937 gen(rand());
    std::uniform_real_distribution<> dis(-3.142, 3.142);
    return dis(gen);
}

int main(int argc, char **argv) {
    setlocale(LC_ALL, "");
    ros::init(argc, argv, "random_track");
    ros::NodeHandle n;

    ros::Publisher publisher = n.advertise<geometry_msgs::Twist>("/turtle1/cmd_vel", 100);
    geometry_msgs::Twist twist;
    twist.linear.x = 3.3;

    ros::Rate rate(2);
    while (ros::ok()) {
        twist.angular.z = generate_random();
        publisher.publish(twist);
        ros::spinOnce();
        rate.sleep();
    }
    return 0;
}

File: src/tf_turtle/scripts/rand_turtle1.py
from random import uniform

import rospy
from geometry_msgs.msg import Twist


def main():
    rospy.init_node("random_track")
    publisher = rospy.Publisher("turtle1/cmd_vel", Twist, queue_size=100)

    twist = Twist()
    twist.linear.x = 3

    rate = rospy.Rate(2)
    while not rospy.is_shutdown():
        twist.angular.z = uniform(-3.142, 3.142)
        publisher.publish(twist)
        rate.sleep()


if __name__ == "__main__":
    main()
